union/union.js

```diff
@@ -0,0 +1,27 @@
+//Iteration 1 - without reduce:
+const union = (...arrays) => {
+    let uniqueNumbers = new Set();
+
+    for (let i=0; i<arrays.length; i++) {
+        let eachArray = arrays[i];
+        for (let ii=0; ii<eachArray.length; ii++) {
+            uniqueNumbers.add(eachArray[ii]);
+            
+        }
+    }
+    return Array.from(uniqueNumbers);
+
+};
+
+//Iteration 2 - with reduce 
+const union = (...arrays) => {
+    let uniqueNumbers = new Set();
+
+    let finder = arrays.reduce((acc, val) => {
+        return uniqueNumbers.has(...val) ? acc : uniqueNumbers.add([val]);
```

Comment:
When you do `...val`, this means to me you're passing in an array to `has`. I'd expect `has` to take a single element. And looking at the documentation, that is indeed the case:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has

So this looks pretty good. The only thing I'm thinking is maybe `add` is not returning the updated set correctly? Also, `add` appears to take in a single element too.

The thing with `reduce` is `acc` is updated with the return value of each iteration, so if the iteration is ever undefined, that will mess it up the entire rest of the way. I'd suggest putting in some `console.log` statements.